src/context.cpp

```diff
@@ -521,11 +522,55 @@ namespace librealsense
             software_sensor::open( profiles );
         }
 
+        void start( frame_callback_ptr callback ) override
+        {
+            //For each stream in sensor_base::get_active_streams() set callback function that will call on_video_frame
+            for ( auto & profile : sensor_base::get_active_streams() )
+            {
+                //stream is the dds_stream matching the librealsense active stream
+                auto & stream = _streams[sid_index( profile->get_unique_id(), profile->get_stream_index() )];
+                stream->start_streaming( [p = profile, this]( const realdds::topics::device::image & dds_frame ) {
+                    rs2_stream_profile prof = { p.get() };
+                    rs2_software_video_frame rs2_frame;
+                    
+                    //Copying from dds into LibRS space, same as copy from USB backend.
+                    //TODO - use memory pool or some other frame allocator
+                    rs2_frame.pixels = new uint8_t[dds_frame.size];
+                    if ( !rs2_frame.pixels )
+                        throw std::runtime_error( "Could not allocate memory for new frame" );
+                    memcpy( rs2_frame.pixels, dds_frame.raw_data.data(), dds_frame.size );
+                    
+                    rs2_frame.deleter = []( void * ptr) { delete[] ptr; };
```

Comment:
Who will delete the `rs2_frame.pixels` allocation? 
Did you check to memory leak during streaming?
Please verify with task manager memory is stable during streaming and update.

Reply:
The deleter is called after the image is displayed (using `frame_continuation` in `on_video_frame`)
I verified that memory is stable for long run (~15 minutes) before opening the PR